raiden/tests/unit/test_tokennetwork.py

```diff
@@ -115,3 +120,114 @@ def test_channel_settle_must_properly_cleanup():
     token_network_state_after_settle = channel_settled_iteration.new_state
     ids_to_channels = token_network_state_after_settle.channelidentifiers_to_channels
     assert channel_state.identifier not in ids_to_channels
+
+
+def test_multiple_channel_states(chain_state, payment_network_state, token_network_state):
+    open_block_number = 10
+    pseudo_random_generator = random.Random()
+    pkey, address = factories.make_privkey_address()
+
+    amount = 30
+    our_balance = amount + 50
+    channel_state = factories.make_channel(
+        our_balance=our_balance,
+        partner_balance=our_balance,
+        partner_address=address,
+    )
+
+    channel_new_state_change = ContractReceiveChannelNew(
+        token_network_state.address,
+        channel_state,
+    )
+
+    channel_new_iteration = token_network.state_transition(
+        token_network_state,
+        channel_new_state_change,
+        pseudo_random_generator,
+        open_block_number,
+    )
+
+    lock_amount = 30
+    lock_expiration = 20
+    lock_secret = sha3(b'test_end_state')
+    lock_secrethash = sha3(lock_secret)
+    lock = HashTimeLockState(
+        lock_amount,
+        lock_expiration,
+        lock_secrethash,
+    )
+
+    mediated_transfer = make_receive_transfer_mediated(
+        channel_state,
+        pkey,
+        1,  # nonce
```

Comment:
for the future, avoid comments for this purpose. Instead either do something like:

```
nonce = 1
amount = 0
function_call(nonce, amount)
```

or
 ```
function_call(nonce=1, amount=0)  # use the keyword arguments
```